### ISSUE TITLE
gzip/xz container image tarball

### ISSUE THREAD

**Issue by pbabinca** [projectatomic/atomic-reactor#191]
Save disk space and speedup data transfers by gzipping container image tarball.

[Koji buildsystem uses xz](https://git.fedorahosted.org/cgit/koji/tree/builder/kojid#n3201) to compress base (container) images which it creates If atomic reactor is used together with koji it would make a sense to use same format for layered tarballs too.

I *believe* it does make a sense to compress this by atomic-reactor as we will gain most advantages as described in first paragraph there.

**Comment by TomasTomecek** [comment]
Is this the workflow?

1. squash layers
2. output to tar
3. compress the tar
4. send it to NFS

If so, I guess that we could compress the tarball right away since we do that for pulp already:

https://github.com/DBuildService/atomic-reactor/blob/master/atomic_reactor/plugins/post_push_to_pulp.py#L136

**Comment by pbabinca** [comment]
Basically yes. I *believe* some owners (like Fedora) would like to use different format than gzip - xz.

**Comment by goldmann** [comment]
You could also file a RFE for the format of the output file in docker-scripts if this would help.

**Comment by bkabrda** [comment]
I'm not sure I understand this properly, but AFAICS we already compress the squashed tarball when pushing to pulp: [1]. Is this what is requested here (possibly with the ability to choose different compression method)?
Alternatively, I could split this into a separate plugin that would do the tarball compression and then another plugin, like pulp would just take the compressed image and push it.
@pbabinca which alternative would you prefer? I'd probably go for creating a separate compression plugin, since various "push-type" plugins could then use the compressed tarball. Also, I'd like to ask @twaugh for comment, since you implemented the `post_push_to_pulp` plugin (AFAICT).

[1] https://github.com/projectatomic/atomic-reactor/blob/master/atomic_reactor/plugins/post_push_to_pulp.py#L136

**Comment by twaugh** [comment]
We dispatch the image tarball in two places: pulp, and (currently) an NFS mount. Only the pulp tarball is compressed at the moment.

We want both outputs to be compressed, and of course it makes sense to do the compression only once. A separate plugin makes sense to me.
